data/json/flags.json

```diff
@@ -64,6 +64,15 @@
     "description": "Blinds the wearer while worn, and provides nominal protection vs flashbang flashes.",
     "info": "This gear <bad>prevents</bad> you from <info>seeing</info> anything."
   },
+  {
+    "id": "EFFECT_NIGHT_VISION",
+    "type": "json_flag",
```

Comment:
The data loaded from type "json_flag" is not used by the effect system. Have you tried that different values within this object will yield different output from the game?

Reply:
This PR adds this new flag type, moving toward the effect system using it more instead of the effect id is the point of the PR?

Reply:
I get the purpose of the PR, however I doubt that this JSON data here (the entry in this file) has any impact on the game.

Where in the game will a player see the content of this entry? Does this somehow change the display in the effect list in the `@` character information screen? I assumed that "json_flag" are only used by the item information dialog.

Reply:
This flag can be used by anything that can cause effect to provide night vision in a way that also displays on the @ screen yes.

Reply:
`json_flag::load` does not load  "properties" nor "context". So how can those values be used within the game?

Your new function `Creature::has_effect_with_flag` is only called from `has_nv`, but not from the code that display character information (include the code that displays current effects), and the ` effect_type::flags` set is protected any not exported to that code. That part of the program does not know that effects now have flags. It therefor wont check any `json_flag` definition.

I tested this branch and changed the "description" of this JSON entry to "FOOBAR", but I could not find any "FOOBAR" string within the game. So where is it actually displayed? Also changing `"color": "green"` to `"color": "foobar"` (certainly an invalid color) does not yield any debug message or error.

Reply:
yes, the properties as an attempt at future proofing that isnt part of this PR, so I will remove that. The context and description I just followed the conention of the other examples. You are saying that json_flag does not use context? Even though all the other ones here has that?

Reply:
> You are saying that json_flag does not use context? 

Nope, this is where it's loaded: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/flag.cpp#L18 and there  is no mention of "context" there.

> Even though all the other ones here has that?

Yes. The file has it since it was first created (around 841109ad6544fd3659a41a21021f31f9f0501931), but it was never loaded into the program.

Reply:
So should I just remove all contexts then?

Reply:
> So should I just remove all contexts then?

If you want to, do so in another PR as it has nothing to do with the content of this PR.

Reply:
I appreciate your CRs, many have been helpful and educational, but I don't really understand what you want from me here. To answer the first question you put, only the FLAG name and the fact that it is a 'json_flag' matters. The flag wont appear in game, because the whole point is that the in game parts of the effect, the flavor, goes on new effects that uses this flag. This way many different flavors can have the same effect.

Reply:
I'm not opposed to the PR itself. Adding flags to effect types is fine.

 I'm just opposed to this single JSON entry because *it does not do anything*. Entries with "json_flag" type apply to *item* flags, but not to effect type flags, so this will show something in the item information window, but it won't affect the display of effects. What does it have to do with this PR (which concerns flags of effect types)?

This entry affects the display of *items* that have the "EFFECT_NIGHT_VISION" flag, but you did not add any such item nor are there any such items in the game right now.

----

In other words: this JSON entry does not affect the game at all. It is about as useful as this:
```JSON
{
    "id": "FOOBAR_DDDD",
    "type": "json_flag",
    "context": [ "SOMETHING....." ]
}
```

It's essentially dead data. *Why* is it there?

Reply:
I added the context to follow the convention of existing JSON FLAGS. This flag is not suppose to be used with items, just Effects, hence the context of EFFECT and absence of item flags. 

My idea is that a single flag could be combined for items and spells that give the same effect, but I don't know enough at this point on how the item system work. There would have to be a even of "puts item on" that you could catch and add the effect with this flag on the player. This seems beyond the scope of this PR though.